packages/mobile/src/screens/change-email-screen/SubScreens.tsx

```diff
@@ -0,0 +1,78 @@
+import { useEffect } from 'react'
+
+import { useField } from 'formik'
+import { useAsync } from 'react-use'
+
+import { Box, Text } from '@audius/harmony-native'
+import { HarmonyTextField, PasswordField } from 'app/components/fields'
+import LoadingSpinner from 'app/components/loading-spinner'
+import { audiusBackendInstance } from 'app/services/audius-backend-instance'
+
+import { SubScreen } from '../change-password-screen/SubScreen'
+import { SubScreenHeader } from '../change-password-screen/SubScreenHeader'
+
+const messages = {
+  changeYourEmail: 'Change Your Email',
+  confirmPasswordHelp: 'Please enter your current email and password.',
+  currentEmail: 'Current Email',
+  currentPassword: 'Current Password',
+  newEmailHelp: 'Enter the new email you would like to use on Audius.',
+  newEmail: 'New email'
+}
+
+export const CurrentEmail = () => {
+  const [{ value: oldEmail }, , { setValue: setOldEmail }] =
+    useField('oldEmail')
+  // Load the email for the user
+  const emailRequest = useAsync(audiusBackendInstance.getUserEmail)
+  useEffect(() => {
+    if (emailRequest.value) {
+      setOldEmail(emailRequest.value)
+    }
+  }, [emailRequest.loading, emailRequest.value, setOldEmail])
+  return (
+    <Text variant='body' size='m'>
+      {oldEmail !== '' ? (
+        oldEmail
+      ) : (
+        <LoadingSpinner style={{ width: 16, height: 16 }} />
+      )}
+    </Text>
+  )
+}
+
+export const ConfirmPasswordSubScreen = () => {
+  return (
+    <SubScreen>
+      <SubScreenHeader
+        title={messages.changeYourEmail}
+        description={messages.confirmPasswordHelp}
+      />
+      <Box>
+        <Text variant='label' size={'xs'}>
+          {messages.currentEmail}
+        </Text>
+        <CurrentEmail />
+      </Box>
+      <PasswordField
+        name='password'
+        label={messages.currentPassword}
+        autoComplete={'password'}
+        clearErrorOnChange={false}
+        helperText={undefined}
+      />
+    </SubScreen>
+  )
+}
+
+export const NewEmailSubScreen = () => {
+  return (
+    <SubScreen>
+      <SubScreenHeader
+        title={messages.changeYourEmail}
+        description={messages.newEmailHelp}
+      />
+      <HarmonyTextField name='email' label={messages.newEmail} />
```

Comment:
we do have "newemailfield" which i think does some good validations, lmk if that doesnt work for you

Reply:
Couple of reasons I didn't use it:
- The hint points to sign in, which doesn't make sense here
- Keeping redux up to date seems out of scope
- Didn't want to couple this to signup toooo much

I did notice I was missing some of the props though, added those. Lmk what you think!